src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessGroupUtils.java

```diff
@@ -0,0 +1,156 @@
+package uk.gov.hmcts.ccd.domain.service.common;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.ccd.domain.model.definition.AccessTypeRoleDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseAccessGroup;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseAccessGroupForUI;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CaseAccessGroupUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CaseAccessGroupUtils.class);
+    public static final String CASE_ACCESS_GROUPS = "CaseAccessGroups";
+
+    protected static final String CCD_ALL_CASES = "CCD:all-cases-access";
+    protected static final String ORGANISATION_POLICY_FIELD = "OrganisationPolicyField";
+    protected static final String ORGANISATION = "Organisation";
+    protected static final String ORGANISATIONID = "OrganisationID";
+    protected static final String ORG_IDENTIFIER_TEMPLATE = "$ORGID$";
+    protected static final String ORG_POLICY_CASE_ASSIGNED_ROLE = "OrgPolicyCaseAssignedRole";
+    protected static final String CASE_ACCESS_GROUP_TYPE = "caseAccessGroupType";
+
+
+    public void updateCaseAccessGroupsInCaseDetails(CaseDetails caseDetails, CaseTypeDefinition caseTypeDefinition) {
```

Comment:
I would expect to see a test class for this given how much logic there is in here.